### PR TITLE
test: migrate reqresp unit tests to vitest

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Consolidate the testing frameworks and migrate to vitest.

**Description**

Migrate `reqresp` unit tests to vitest.

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/reqresp/src/encoders/responseEncode.ts [hunk 1]
```diff
@@ -28,12 +28,10 @@ export function responseEncodeSuccess(
...
       // <context-bytes> - from altair
       const contextBytes = getContextBytes(protocol.contextBytes, chunk);
-      if (contextBytes) {
-        yield contextBytes as Buffer;
-      }
+      if (contextBytes) yield Buffer.from(contextBytes);
 
       // <encoding-dependent-header> | <encoded-payload>
-      yield* writeEncodedPayload(chunk.data, protocol.encoding);
+      yield* writeEncodedPayload(Buffer.from(chunk.data), protocol.encoding);
```
Comment (wemeetagain) [inline packages/reqresp/src/encoders/responseEncode.ts]: why? this PR shouldn't change `src/` unless absolutely necessary

**Review by wemeetagain — CHANGES_REQUESTED** [review]

File: packages/reqresp/src/encoders/responseEncode.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/reqresp/src/encoders/responseEncode.ts]: The `chai` assertions were passing on `Buffer` and `Uint8Array` comparison while the `vitest` is not. 

So I tried to convert all tests fixture to `Uint8Array` and then had to refactor some code to accept that type of data. 

Will go over once again to make sure nothing unnecessary change is added here.

**Comment by nazarhussain** [comment]
@wemeetagain Please re-review this PR. I went through the changes again and made sure only the absolute necessary changes are included from the `src`.

**Review by g11tech — COMMENTED** [review]
looks good, would let @wemeetagain decided on Buffer.from usage on the src/responseEncode.ts, afaik we didn't want any actual Buffer usage in the main src

File: packages/reqresp/src/encoders/responseEncode.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export function responseEncodeSuccess(
       // <context-bytes> - from altair
       const contextBytes = getContextBytes(protocol.contextBytes, chunk);
       if (contextBytes) {
-        yield contextBytes as Buffer;
+        yield Buffer.from(contextBytes);
```
Comment (wemeetagain) [inline packages/reqresp/src/encoders/responseEncode.ts]: I don't think this is necessary?

**Review by nflaig — APPROVED** [review]
Looks good to me, PR no longer modifies `src/`

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.14.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.14.0) :tada:

**Pull request merged**